### ISSUE TITLE
Error during default engine config test and fuel consumption test

### ISSUE THREAD

**Issue by kimjarvis** [tessi/wasmex#405]
Failing tests 
On master branch and v0.8.2 tag.  
Observed on MacOS 13.1 and Ubuntu 22.04. 

```
iex -S mix test
```

```
 1) test error handling handles errors occuring during Wasm execution with default engine config (WasmexTest)
     test/wasmex_test.exs:264
     Assertion with == failed
     code:  assert Wasmex.call_function(pid, :divide, [1, 0]) == {:error, String.trim(expected_reason)}
     left:  {
              :error,
              "Error during function excecution: `error while executing at wasm backtrace:\n    0: 0x394c - <unknown>!__rust_start_panic\n    1: 0x391a - <unknown>!rust_panic\n    2: 0x38ea - <unknown>!std::panicking::rust_panic_with_hook::hbafe3e603d331223\n    3: 0x2fa8 - <unknown>!std::panicking::begin_panic_handler::{{closure}}::h8ab6ee68d5b4c391\n    4: 0x2f0a - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::h008f69666d134159\n    5: 0x3541 - <unknown>!rust_begin_unwind\n    6: 0x3c32 - <unknown>!core::panicking::panic_fmt::h1d17fc068f528130\n    7: 0x3c91 - <unknown>!core::panicking::panic::h27b5eefa3e4ff738\n    8:  0x5f4 - <unknown>!divide`."
            }
     right: {
              :error,
              "Error during function excecution: `error while executing at wasm backtrace:\n    0: 0x395d - <unknown>!__rust_start_panic\n    1: 0x3951 - <unknown>!rust_panic\n    2: 0x3921 - <unknown>!std::panicking::rust_panic_with_hook::he04cb00575f2a1e3\n    3: 0x2fc0 - <unknown>!std::panicking::begin_panic_handler::{{closure}}::hb733f0aa505760cf\n    4: 0x2f25 - <unknown>!std::sys_common::backtrace::__rust_end_short_backtrace::h6beefa0bcab220bc\n    5: 0x3575 - <unknown>!rust_begin_unwind\n    6: 0x3c47 - <unknown>!core::panicking::panic_fmt::h586d720a90aa8503\n    7: 0x3c9c - <unknown>!core::panicking::panic::h0859aaac783261b2\n    8:  0x567 - <unknown>!divide`."
            }
     stacktrace:
       test/wasmex_test.exs:288: (test)
```

```
  2) test fuel consumption &Wasmex.call_function/3 with fuel_consumption (WasmexTest)
     test/wasmex_test.exs:310
     Assertion with == failed
     code:  assert Wasmex.call_function(pid, :void, []) ==
              {:error,
               "Error during function excecution: `error while executing at wasm backtrace:\n    0:  0x3bf - <unknown>!void`."}
     left:  {:error, "Error during function excecution: `error while executing at wasm backtrace:\n    0:  0x44d - <unknown>!void`."}
     right: {:error, "Error during function excecution: `error while executing at wasm backtrace:\n    0:  0x3bf - <unknown>!void`."}
     stacktrace:
       test/wasmex_test.exs:320: (test)
```

**Comment by ricochet** [comment]
I ran into this as well. This will happen for different architectures.

**Comment by ricochet** [comment]
Fixed by #402

**Comment by etehtsea** [comment]
> Fixed by #402

Superseded by https://github.com/tessi/wasmex/pull/410

**Comment by tessi** [comment]
Hey @kimjarvis thanks for reporting this issue and apologies for the super late reply. With the help of @etehtsea and @ricochet this got solved and merged in #410 today. I assume this test failure is gone for good - if not, please re-open this ticket :)
